test/modules/olMap/handler/mfp/OlMfpHandler.test.js

```diff
@@ -0,0 +1,224 @@
+			const actualLayer = handler.activate(map);
+
+			expect(actualLayer).toBeTruthy();
+			expect(handler._registeredObservers).toHaveSize(5);
+		});
+
+		it('initializing mfpBoundaryFeature only once', () => {
+			const map = setupMap();
+			setup();
+
+			const handler = new OlMfpHandler();
+			const mfpBoundaryFeatureSpy = spyOn(handler._mfpBoundaryFeature, 'setStyle').and.callThrough();
+
+			handler.activate(map); // --> mfpLayer is now initialized
+			const mfpLayerSpy = spyOn(handler._mfpLayer, 'on').withArgs('postrender', jasmine.any(Function)).and.callThrough();
+			handler.activate(map);
+
+			expect(mfpBoundaryFeatureSpy).toHaveBeenCalledOnceWith(thumbnailStyleFunction);
+			expect(mfpLayerSpy).not.toHaveBeenCalled();
+		});
+
+		it('updates mfpPage after store changes', () => {
+			const current = { id: 'bar', scale: 42 };
+			const map = setupMap();
+			setup();
+
+			const handler = new OlMfpHandler();
+			handler.activate(map);
+
+			const updateSpy = spyOn(handler, '_updateMfpPage').withArgs(current).and.callThrough();
+			setCurrent(current);
+
+			expect(updateSpy).toHaveBeenCalled();
+		});
+
+		it('updates mfpPreview after store changes', () => {
+			const center = [0, 0];
+			const map = setupMap();
+			setup();
+
+			const handler = new OlMfpHandler();
+			handler.activate(map);
+
+			const updateSpy = spyOn(handler, '_updateMfpPreview').and.callThrough();
+			changeLiveCenter(center);
+
+			expect(updateSpy).toHaveBeenCalled();
+		});
+
+		it('updates rotation after store changes', () => {
+			const map = setupMap();
+			setup();
+
+			const handler = new OlMfpHandler();
+			const updateSpy = spyOn(handler, '_updateRotation').and.callThrough();
+
+
+			handler.activate(map);
+			changeCenter([0, 42]);
+
+			expect(updateSpy).toHaveBeenCalled();
+			updateSpy.calls.reset();
+
+			changeZoom(2);
+
+			expect(updateSpy).toHaveBeenCalled();
+			updateSpy.calls.reset();
+
+			changeRotation(42);
+			expect(updateSpy).toHaveBeenCalled();
+		});
+	});
+
+	describe('when deactivate', () => {
+		it('unregisters observer', () => {
+			const map = setupMap();
+			setup();
+
+			const handler = new OlMfpHandler();
+			handler.activate(map);
+			const spyOnUnregister = spyOn(handler, '_unregister').withArgs(handler._registeredObservers).and.callThrough();
+			handler.deactivate(map);
+
+			expect(handler._mfpLayer).toBeNull();
+			expect(spyOnUnregister).toHaveBeenCalled();
+		});
+
+
+	});
+
+	describe('_getAzimuth', () => {
+
+		it('calculates the intermediate azimuth for a quadrangle polygon', () => {
+			setup();
+			const classUnderTest = new OlMfpHandler();
+			const nonUniformQuadrangle = new Polygon([[[0, 10], [10, 9], [10, 0], [0, -2], [0, 10]]]);
+			const squaredQuadrangle = new Polygon([[[0, 10], [10, 9], [10, 0], [0, -1], [0, 10]]]);
+
+			expect(classUnderTest._getAzimuth(nonUniformQuadrangle)).toBeCloseTo(0.048863, 4);
+			expect(classUnderTest._getAzimuth(squaredQuadrangle)).toBeCloseTo(0.0, 5);
+		});
+	});
```

Comment:
Should we add also tests for other private methods #_getOptimalScale(), #_getPageLabel(), etc?

Reply:
Yes, i have added tests for `_getOptimalScale()` `_getPageLabel() ` and  `_createMpfBoundary()`.